src/ifaddrs.rs

```diff
@@ -42,11 +42,43 @@ cfg_if! {
     }
 }
 
+/// Workaround a bug in XNU where netmasks will always have the wrong size in
+/// the sa_len field due to the kernel ignoring trailing zeroes in the structure
+/// when setting the field. See https://github.com/nix-rust/nix/issues/1709#issuecomment-1199304470
+///
+/// To fix this, we overwrite `sa_len` to have the known (static) value of the
+/// structure.
+#[cfg(target_os = "macos")]
+fn workaround_xnu_bug(info: &libc::ifaddrs) {
+    if info.ifa_netmask.is_null() {
+        return
+    }
+    let family = unsafe {
+        (*info.ifa_netmask).sa_family
+    };
+
+    let fixup_size = match i32::from(family) {
+        libc::AF_INET => std::mem::size_of::<libc::sockaddr_in>(),
+        libc::AF_INET6 => std::mem::size_of::<libc::sockaddr_in6>(),
+        _ => return
+    };
+
+    assert!(fixup_size < u8::MAX.into());
+    let sock = info.ifa_netmask;
+    unsafe {
+        (*sock).sa_len = (*sock).sa_len.max(fixup_size as u8);
+    }
```

Comment:
```suggestion
    let sock = info.ifa_netmask;
    if sock.is_null() {
        return
    }
    let family = unsafe {
        (*sock).sa_family
    };

    let fixup_size = match i32::from(family) {
        libc::AF_INET => std::mem::size_of::<libc::sockaddr_in>(),
        libc::AF_INET6 => std::mem::size_of::<libc::sockaddr_in6>(),
        _ => return
    };

    assert!(fixup_size < u8::MAX.into());
    unsafe {
        (*sock).sa_len = (*sock).sa_len.max(fixup_size as u8)
    };
```

Uses `sock` everywhere so it's clearer we always look at the same memnber

Reply:
Uh, it's highlighting changes I haven't made ? Sorry for that, I don't know what's happening